File: 251001/Stalbou/publisher/src/main/java/org/ex/distributed_computing/controller/NoticeController.java
package org.ex.distributed_computing.controller;

import jakarta.validation.Valid;
import java.util.List;
import lombok.RequiredArgsConstructor;
import lombok.SneakyThrows;
import org.ex.distributed_computing.dto.request.NoticeRequestDTO;
import org.ex.distributed_computing.dto.response.NoticeResponseDTO;
import org.ex.distributed_computing.service.NoticeService;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/notices")
@RequiredArgsConstructor
public class NoticeController {

  private final NoticeService noticeService;

  @GetMapping
  public List<NoticeResponseDTO> getAllNotices() {
    return noticeService.getAllNotices();
  }

  @GetMapping("/{id}")
  @SneakyThrows
  public ResponseEntity<NoticeResponseDTO> getNoticeById(@PathVariable Long id) {
    Thread.sleep(300);
    return ResponseEntity.ok(noticeService.getNoticeById(id));
  }

  @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
  public ResponseEntity<NoticeResponseDTO> createNotice(@Valid @RequestBody NoticeRequestDTO requestDTO) {
    return ResponseEntity.status(HttpStatus.CREATED).body(noticeService.createNotice(requestDTO));
  }

  @PutMapping
  public ResponseEntity<NoticeResponseDTO> updateNotice(@Valid @RequestBody NoticeRequestDTO requestDTO) {
    return ResponseEntity.ok(noticeService.updateNotice(requestDTO));
  }

  @DeleteMapping("/{id}")
  public ResponseEntity<Void> deleteNotice(@PathVariable Long id) {
    noticeService.deleteNotice(id);
    return ResponseEntity.noContent().build();
  }
}

File: 251001/Stalbou/discussion/src/main/kotlin/org/ex/discussion/controller/NoticeController.kt
package org.ex.discussion.controller

import org.ex.discussion.dto.request.NoticeRequestDTO
import org.ex.discussion.dto.response.NoticeResponseDTO
import org.ex.discussion.service.NoticeService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/notices")
class NoticeController(val noticeService: NoticeService) {

    @GetMapping
    fun getAllNotices(): List<NoticeResponseDTO> = noticeService.getAllNotices()

    @GetMapping("/{id}")
    fun getNoticeById(@PathVariable id: Long): NoticeResponseDTO = noticeService.getNoticeById(id)

    @PostMapping
    fun createNotice(@RequestBody noticeRequest: NoticeRequestDTO): ResponseEntity<NoticeResponseDTO> {
        return ResponseEntity.status(HttpStatus.CREATED).body(noticeService.createNotice(noticeRequest))
    }

    @PutMapping
    fun updateNotice(@RequestBody noticeRequest: NoticeRequestDTO): ResponseEntity<NoticeResponseDTO> {
        println("Received updating notice request $noticeRequest")
        return ResponseEntity.ok(noticeService.updateNotice(noticeRequest))
    }

    @DeleteMapping("/{id}")
    fun deleteNotice(@PathVariable("id") noticeId: Long): ResponseEntity<Void> {
        noticeService.deleteNotice(noticeId)
        return ResponseEntity.ok().build()
    }
}